public/js/components/EditorSearchBar.js

```diff
@@ -0,0 +1,159 @@
+const React = require("react");
+const { DOM: dom, PropTypes } = React;
+const { findDOMNode } = require("react-dom");
+const Svg = require("./utils/Svg");
+const { isEnabled } = require("../feature");
+const { find, findNext, findPrev } = require("../utils/source-search");
+const classnames = require("classnames");
+
+require("./EditorSearchBar.css");
+
+function countMatches(query, text) {
+  const re = new RegExp(query, "g");
+  const match = text.match(re);
+  return match ? match.length : 0;
+}
+
+const EditorSearchBar = React.createClass({
+
+  propTypes: {
+    editor: PropTypes.object,
+    sourceText: PropTypes.object
+  },
+
+  displayName: "EditorSearchBar",
+
+  getInitialState() {
+    return {
+      enabled: false,
+      query: "",
+      count: 0,
+      index: 0
+    };
+  },
+
+  contextTypes: {
+    shortcuts: PropTypes.object
+  },
+
+  setupKeyboardShortcuts() {
+    if (this.keyShortcutsEnabled) {
+      return;
+    }
+
+    this.keyShortcutsEnabled = true;
```

Comment:
"needed the window"? Shouldn't `window` always exist? I encourage you to go down these rabbit holes :) Otherwise we are just accumulating technical debt.